src/test/java/com/jfrog/ide/idea/scan/YarnScannerTest.java

```diff
@@ -0,0 +1,113 @@
+package com.jfrog.ide.idea.scan;
+
+import com.intellij.testFramework.HeavyPlatformTestCase;
+import com.intellij.util.ConcurrencyUtil;
+import com.jfrog.ide.common.deptree.DepTree;
+import com.jfrog.ide.common.nodes.DependencyNode;
+import com.jfrog.ide.common.nodes.FileTreeNode;
+import com.jfrog.ide.common.scan.GraphScanLogic;
+import org.apache.commons.io.FileUtils;
+import org.jetbrains.plugins.gradle.service.project.open.GradleProjectImportUtil;
+import org.jfrog.build.api.util.NullLog;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.*;
+import java.util.concurrent.ExecutorService;
+
+public class YarnScannerTest extends HeavyPlatformTestCase {
+    private static final Path YARN_ROOT = Paths.get(".").toAbsolutePath().normalize().resolve(Paths.get("src", "test", "resources", "yarn"));
+    private ExecutorService executorService;
+    private String tempProjectDir;
+
+    @Override
+    protected void setUp() throws Exception {
+        super.setUp();
+        tempProjectDir = createTempProjectDir("exampleYarnPackage");
+        executorService = ConcurrencyUtil.newSameThreadExecutorService();
+    }
+
+    @Override
+    protected void tearDown() throws Exception {
+        try {
+            executorService.shutdown();
+        } finally {
+            // Ensure that tearDown gets executed even if an exception is thrown
+            super.tearDown();
+        }
+    }
+
+    private String createTempProjectDir(String projectName) throws IOException {
+        // Using a virtual directory allows each test to have its own isolated workspace, preventing interference between tests.
+        String tempProjectDir = getTempDir().createVirtualDir(projectName).toNioPath().toString();
+        FileUtils.copyDirectory(YARN_ROOT.resolve(projectName).toFile(), FileUtils.getFile(tempProjectDir));
+        return tempProjectDir;
+    }
+
+    public void testGetPackageNameToVersionsMap() {
+        YarnScanner yarnScanner = new YarnScanner(getProject(), tempProjectDir, executorService, new GraphScanLogic(new NullLog()));
+        Set<String> packages = new HashSet<>(Arrays.asList("package1:1.0.123", "package2:2.0.9", "package1:1.0.123", "package2:2.1.7"));
+        Map<String, Set<String>> packageNameToVersions = yarnScanner.getPackageNameToVersionsMap(packages);
+
+        assertEquals(2, packageNameToVersions.size());
+        assertTrue(packageNameToVersions.containsKey("package1"));
+        assertTrue(packageNameToVersions.containsKey("package2"));
+        assertFalse(packageNameToVersions.containsKey("package3"));
+
+        assertEquals(new HashSet<>(List.of("1.0.123")), packageNameToVersions.get("package1"));
+        assertEquals(new HashSet<>(Arrays.asList("2.0.9", "2.1.7")), packageNameToVersions.get("package2"));
```

Comment:
We can use Sets.newHashSet of `com.google.common.collect.Sets`
```suggestion
        assertEquals(Sets.newHashSet("1.0.123"), packageNameToVersions.get("package1"));
        assertEquals(Sets.newHashSet("2.0.9", "2.1.7"), packageNameToVersions.get("package2"));
```